blog/2022/storage-map.adoc

```diff
@@ -0,0 +1,108 @@
+supported by the new store, these are only supported by the legacy. We do consider
+second-level support for these databases, based on results of a community survey.
+
+For development and testing purposes, we offer a fast in-memory store called `chm`.
+This store is intentionally not capable of zero-downtime upgrade as that is not needed
+for development. As such, it offers also a quick Keycloak-without-database deployment.
+
+We will be glad to hear your feedback in https://github.com/keycloak/keycloak/discussions/new?category=keycloak-x-new-store&title=%5BUser%20feedback%5D%20&labels=team/storage-sig[GitHub Discussions].
+
+## Try it out
+
+The following section contains examples of how to configure Keycloak 19
+with the new store.
+
+WARNING: The command line options starting with `--storage` might change
+         without prior warning, based on the feedback from the community
+         and project needs.
+
+### See what Keycloak can do without a database
+
+After downloading, you can try Keycloak in a single-node deployment as simply as running
+
+   bin/kc.sh start-dev --storage=chm
+
+### Next step: Persist data in relational database
+
+_Requirements:_ Postgres 14 or CockroachDB 22.1 (or newer)
+
+To keep the data in a database, run the following command:
+
+   bin/kc.sh start-dev --storage=jpa --db-url=<jdbc-url> --db-username=<username> --db-password=<password>
+
+Reference to https://github.com/keycloak/keycloak/discussions/9596[Q&A live document].
+
+CAUTION: This feature has known issues in CockroachDB, see https://github.com/keycloak/keycloak/issues/11618[this tracker for list].
+
+### Alternative next step: Persist data in external Infinispan
+
+_Requirements:_ Infinispan 12
+
+If you have a cluster of running Infinispan, you can use it for storing the data as well,
+just run the following command:
+
+   bin/kc.sh start-dev --storage=hotrod --storage-hotrod-host=<host> --storage-hotrod-port=11222 --storage-hotrod-username=<username> --storage-hotrod-password=<password>
+
+Reference to https://github.com/keycloak/keycloak/discussions/12010[Q&A live document].
+
+=== Next step: Store data in separate storages
+
+Eventually it turns out that some of the data should be stored in files or a database, other
+(e.g. session) data should be stored in Infinispan. Can this be achieved?
+
+Yes! Keycloak storage is divided into the following storage areas: Realms,
+clients, users, groups, roles, client scopes, authorization services,
+events, authentication sessions, user/client sessions, login failures,
+and single-use tokens.
+
+Each area maintains its own storage for storing the data, and each area
+is independent of others. In other words, realms can be served by database and
+users by Infinispan.
+
+To have all the session data stored in external Infinispan,
+and realm / client / user / group / role / ... data stored in a relational database, you
+can issue the following command:
+
+   bin/kc.sh start-dev --storage=hotrod \
```

Comment:
Should this be `jpa` instead of `hotrod`? 

```suggestion
   bin/kc.sh start-dev --storage=jpa \
```

Reply:
Good catch, thanks!